design-proposals/guest-agent-exec.md

```diff
@@ -0,0 +1,56 @@
+## Virtctl Guest Agent Exec (ga-exec)
+
+**Authors:** Ryan Hallisey NVIDIA
+
+## Motivation
+The guest-agent is a powerful tool for querying a guest, but it isn’t easy for
+a user to do from kubectl.  To run a guest-agent command, the user has to locate
+a launcher pod, exec into the pod, run the guest-agent-exec command, capture the pid
+from the result, and run a second command - guest-agent-status - to get the result.  This
+process can be simplified.
+
+KubeVirt already has a lot of code using the guest-agent, so the idea for this proposal is
+to extend the existing connections to the guest-agent all the way up to the virtctl client
+so users can have ease of access too.
+
+#### Goals
+* Provide an client command for the user to run qemu guest-agent commands and return the output to stdout
+* Reduce the number of client commands it takes to run a guest-agent command
+
+#### Non-Goals
+* Allowing a user to open a shell in a guest
+
+#### Use-cases
```

Comment:
Do you have a use-case where you can't use `ssh` which you want to solve?

Reply:
Ya if the guest has ssh login disabled, then I would need to rely on the agent.